src/main/java/org/prebid/server/bidder/rubicon/RubiconBidder.java

```diff
@@ -972,14 +979,43 @@ private static void copyLiveintentSegment(Map<String, List<ExtUserEid>> sourceTo
         }
     }
 
-    private void mergeFirstPartyDataFromUser(User user, ObjectNode result) {
+    private void mergeFirstPartyDataFromUser(ExtUser userExt, ObjectNode result) {
         // merge OPENRTB.user.ext.data.* to XAPI.user.ext.rp.target.*
-        final ExtUser userExt = user != null ? user.getExt() : null;
         if (userExt != null) {
             populateFirstPartyDataAttributes(userExt.getData(), result);
         }
     }
 
+    private void enrichWithIabAttrubute(ObjectNode target, List<Data> data) {
+        if (shouldBeCopiedSegmentIds(data)) {
+            final ArrayNode iab = target.putArray("iab");
+            addIdsToIabAttribute(iab, data);
+        }
+    }
+
+    private boolean shouldBeCopiedSegmentIds(List<Data> data) {
+        return CollectionUtils.isNotEmpty(data) && data.stream()
```

Comment:
My suggestion here is to avoid conversion, so we can use `JsonNode` object:
```
return CollectionUtils.emptyIfNull(data).stream()
                .map(Data::getExt)
                .filter(Objects::nonNull)
                .map(ext -> ext.get("taxonomyname"))
                .filter(Objects::nonNull)
                .filter(JsonNode::isTextual)
                .map(JsonNode::textValue)
                .anyMatch(value -> StringUtils.containsIgnoreCase(value, "iab"));
```
Then `RubiconDataExt` will be not required.

Also, we cannot guaranteed `taxonomyname` is text field in case of FPD.
Pls add unit test for other types of this field like array, object, etc.